### ISSUE TITLE
Dummy model failing when comparing across more than one year 

### ISSUE THREAD

**Issue by lewisblake** [metno/pyaerocom#732]
I am working on the obs-only experiment and trying to read in `concpm10` data for 2019 and 2020 from `EEAAQeRep.v2`. Both of these years work independently. 

This is demonstrable by the fact that both the [2021-reporting has the 2019 data](https://aeroval.met.no/evaluation.php?project=emep&exp_name=2021-reporting&station=WORLD&par=concpm10#) and the [2022-reporting has the 2020 data](https://aeroval.met.no/evaluation.php?project=emep&exp_name=2022-reporting&station=AT0ENK1&par=concpm10#). 

However, when I try to run the experiment with both years, it crashes with the following error:

```
2022-09-13 11:10:32:pyaerocom.io.iris_io:WARNING:Failed to load /home/lewisb/MyPyaerocom/tmp/dummy_model/renamed/aerocom3_dummy_model_concpm10_Surface_2019_monthly.nc. Reason: Traceback (most recent call last):
  File "/lustre/storeB/users/lewisb/Python/Evaluations/pyaerocom/pyaerocom/io/iris_io.py", line 331, in _check_correct_time_dim
    check_time_coord(cube, ts_type, year)
  File "/lustre/storeB/users/lewisb/Python/Evaluations/pyaerocom/pyaerocom/io/iris_io.py", line 415, in check_time_coord
    raise UnresolvableTimeDefinitionError(
pyaerocom.exceptions.UnresolvableTimeDefinitionError: Expected 12 timestamps but data has 24

```

Basically, a mismatch in the length of the of the cubes. In `pyaerocom/iris_io.py`:

```python
num_per = len(tidx)
    num = len(tdim.points)

    if not num == num_per:
        if tidx[0].is_leap_year:
            if not _check_leap_year(num, num_per, ts_type):
                raise UnresolvableTimeDefinitionError(
                    f"Expected {len(tidx)} timestamps but data has {num}"
                )

        else:
            raise UnresolvableTimeDefinitionError(
                f"Expected {len(tidx)} timestamps but data has {num}"
            )

```

Note that for this experiment I have been able to do similar things against EBAS.
